adafruit_pypixelbuf.py

```diff
@@ -164,24 +154,35 @@ def bpp(self):
     def brightness(self):
         """
         Float value between 0 and 1.  Output brightness.
-        If the PixelBuf was allocated with two both a buf and a rawbuf,
-        setting this value causes a recomputation of the values in buf.
-        If only a buf was provided, then the brightness only applies to
-        future pixel changes.
-        In DotStar mode
+
+        When brightness is less than 1.0, a second buffer will be used to store the color values
+        before they are adjusted for brightness.
         """
         return self._brightness
 
     @brightness.setter
     def brightness(self, value):
-        self._brightness = min(max(value, 0.0), 1.0)
+        value = min(max(value, 0.0), 1.0)
+        change = value - self._brightness
+        if -0.001 < change < 0.001:
+            return
+
+        self._brightness = value
+
+        if self._pre_brightness_buffer is None:
+            self._pre_brightness_buffer = bytearray(
+                (i for i in self._post_brightness_buffer)
```

Comment:
Isn't this just a copy of another byte array? You could just do `bytearray(self._post_brightness_buffer)`.

Reply:
Wow, well that is certainly simpler, thanks for pointing that out! I knew something about the way I did it felt wrong, but I couldn't find the way to copy a bytearray. I thought maybe `self._post_brightness_buffer.copy()` would work but when it didn't I went with the generator.